File: backend/app/domain/repositories/care_log_repository.py
from abc import ABC, abstractmethod
from datetime import date
from uuid import UUID

from app.domain.entities.care_log import CareLog


class CareLogRepository(ABC):
    @abstractmethod
    async def get(self, id: UUID) -> CareLog | None:
        ...

    @abstractmethod
    async def get_by_baby_and_date(self, baby_id: UUID, target_date: date) -> list[CareLog]:
        ...

    @abstractmethod
    async def save(self, record: CareLog) -> CareLog:
        ...

    @abstractmethod
    async def update(self, record: CareLog) -> CareLog:
        ...

    @abstractmethod
    async def delete(self, id: UUID) -> None:
        ...

File: zzippu_ios/zzippu/Domain/Repositories/CareLogRepository.swift
// Domain/Repositories/CareLogRepository.swift
// Foundation only — SwiftUI/SwiftData import 금지

import Foundation

protocol CareLogRepository {
    func create(_ log: CareLog) async throws -> CareLog
    func update(_ log: CareLog) async throws -> CareLog
    func delete(id: UUID, babyId: UUID) async throws
    func list(babyId: UUID, on day: Date) async throws -> [CareLog]
}
